link_fixer.py

```diff
@@ -440,21 +476,29 @@ def get_synced_item(item_id, project_id):
                     hyperlink_string = start_link + encoded_name + end_link
                     value = value.replace(hyperlink_string, corrected_hyperlink_string)
 
-            # we have a bad link for this item?
-            if bad_link_found:
-                # let's build out an object of all the data we car about for patching and logging
-                broken_link_data = {
-                    'fieldName': key,
-                    'newValue': value,
-                    'oldValue': original_value,
-                    'counter': str(bad_link_count)
-                }
-                broken_list = broken_link_map.get(item_id)
-                if broken_list is None:
-                    broken_list = [broken_link_data]
-                else:
-                    broken_list.append(broken_link_data)
-                broken_link_map[item_id] = broken_list
+                # we have a bad link for this item?
+                if bad_link_found:
+                    # Before we replace the hyperlink, let's check if it's locked and log it to Excel if so
+                    if item_lock_dict['locked']:
+                        print("Item was locked and has a broken link.  Logging to Excel File...\n")
+                        log_locked_items(workbook, name_field_for_excel, str(item_id), key, str(hyperlink))
+
+                    # let's build out an object of all the data we care about for patching and logging
+                    else:
+                        broken_link_data = {
+                            'fieldName': key,
+                            'newValue': value,
+                            'oldValue': original_value,
+                            'counter': str(bad_link_count)
+                        }
+                        broken_list = broken_link_map.get(item_id)
+                        if broken_list is None:
+                            broken_list = [broken_link_data]
+                        else:
+                            broken_list.append(broken_link_data)
+                        broken_link_map[item_id] = broken_list
+
+    workbook.save("locked_items.xlsx")
```

Comment:
Move save workbook to after post phase.

Check any failed patch response for the word locked in the reason.  and add to workbook.

Reply:
may need to add "item Name" to "broken_link_data" for access in post phase